rasa/core/actions/action.py

```diff
@@ -1117,7 +1129,9 @@ async def _execute_validation_action(
         # candidate we assume that it was valid. The custom action has to return a
         # SlotSet(slot_name, None) event to mark a Slot as invalid.
         return validate_events + [
-            event for event in slot_events if event.key not in validated_slot_names
+            event
+            for event in slot_events
+            if isinstance(event, SlotSet) and event.key not in validated_slot_names
```

Comment:
Because `validate_events` is of type `List[Event]` it won't be able to concatenate with an event of type `SlotSet`, because of the same `operator` type error as above 🤔